### ISSUE TITLE
Behaviors of AveragePooling2D((x, x), padding="same") are different between Keras and WebDNN

### ISSUE THREAD

**Issue by y-ich** [mil-tokyo/webdnn#694]
Hi.

on Keras, AveragePooling2D((2, 2), padding="same") outputs an tensor with 1 element when input's dimension is same as pool_size.
on WebDNN, it outputs an tensor(vector) with 4 elements. The first elements may match Keras' one.

on Keras, AveragePooling2D((3, 3), padding="same") outputs an tensor with 1 element when input's dimension is same as pool_size.
on WebDNN, it outputs an tensor(tensor) with 4 elements. No elements match Keras' one.

Larger padding_size seems same as the case of (3, 3).

Padding="same" seems a key.

Here is a reproduction code.
https://github.com/y-ich/averagepool2d_reproduction

**Comment by Kiikurage** [comment]
Thank you for your report. I succeed to reproduce the problem. I'll fix it soon.

**Comment by Kiikurage** [comment]
I fixed the problem in #695

**Comment by y-ich** [comment]
Thank you!

Now the reproduction code shows that tensor sizes became same but values of an element are still different between Keras and WebDNN in the case of size = 3.
On Keras, result is 1.0, but result is 0.4444444477558136 on WebDNN on my computer.

If it is due to different padding algorithms between Keras and WebDNN, just please let me know it.

**Comment by Kiikurage** [comment]
Yes, this is due to different computing algorithm for average pooling.

In case of `size=3, padding="SAME"`, padded input tensor is like follows,

```text
+-+-+-+-+-+
|0|0|0|0|0|
+-+-+-+-+-+
|0|1|1|1|0|
+-+-+-+-+-+
|0|1|1|1|0|
+-+-+-+-+-+
|0|1|1|1|0|
+-+-+-+-+-+
|0|0|0|0|0|
+-+-+-+-+-+
```

and first input window is like follows,

```text
+-+-+-+
|0|0|0|
+-+-+-+
|0|1|1|
+-+-+-+
|0|1|1|
+-+-+-+
```

Keras ignore padded elements, so result is `4 / 4 = 1`.

WebDNN consider about all elements, so result is `4 / 9 = 0.44..`.

**Comment by y-ich** [comment]
Thank you for your explanation.

As I am porting neural networks for board game Go, not for image processing, I am afraid how this difference effects, but it is another business^^
